crates/ruff/src/checkers/imports.rs

```diff
@@ -54,5 +55,45 @@ pub fn check_imports(
             &blocks, python_ast, locator, stylist, settings, autofix,
         ));
     }
-    diagnostics
+    let mut imports: FxHashMap<PathBuf, Vec<Import>> = FxHashMap::default();
+    let mut imports_vec = vec![];
+    for &block in &blocks {
+        block.imports.iter().for_each(|&stmt| match &stmt.node {
+            StmtKind::Import { names } => {
+                // from testing, seems this should only have one entry
+                imports_vec.push(Import {
+                    name: names[0].node.name.clone(),
+                    location: stmt.location,
+                    end_location: stmt.end_location.unwrap(),
+                });
+            }
+            StmtKind::ImportFrom { module, names, .. } => imports_vec.extend(
+                names
+                    .iter()
+                    .map(|name| Import {
+                        name: format!(
+                            "{}.{}",
+                            {
+                                if let Some(n) = module {
+                                    n
+                                } else {
+                                    ""
+                                }
```

Comment:
Nit: Does this work
```suggestion
                                module.unwrap_or_default()
```

Reply:
I'll give it a go

Reply:
Hm, I think that as the `Default` trait isn't implement for `std::string::String` we can't _just_ do this, but I can do something similar.